client/overview/task-list/tasks.js

```diff
@@ -20,15 +23,57 @@ export const getTasks = ( {
 	isAccountOverviewTasksEnabled,
 	numDisputesNeedingResponse = 0,
 } ) => {
-	const { status, currentDeadline, pastDue, accountLink } = accountStatus;
+	const renderModal = (
+		errorMessages,
+		status,
+		accountLink,
+		currentDeadline
+	) => {
+		const container = document.createElement( 'div' );
+		container.id = 'wcpay-update-business-details-container';
+		document.body.appendChild( container );
+		ReactDOM.render(
+			<UpdateBusinessDetailsModal
+				errorMessages={ errorMessages }
+				accountStatus={ status }
+				accountLink={ accountLink }
+				currentDeadline={ currentDeadline }
+			/>,
+			container
+		);
+	};
+
+	const getErrorMessagesFromRequirements = ( requirements ) => {
+		if ( 'errors' in requirements ) {
+			const errorMessages = requirements.errors.map( ( error ) => {
+				return error.reason;
+			} );
+			return [ ...new Set( errorMessages ) ];
+		}
+
+		return [];
+	};
```

Comment:
When we define a pure function like this one, we can move it outside the exported one, or even in a different file, if that help us organize better the code 🙂 

As a bonus tip, we can rewrite it using new ES features to make it shorter and hopefully more readable:
```js
	const getErrorMessagesFromRequirements = ( requirements ) => [
		...new Set( requirements?.errors?.map( ( error ) => error.reason ) ),
	];
```

Although, as you already provided a fallback value for `requirements.errors` in the PHP side: https://github.com/Automattic/woocommerce-payments/blob/114b69231ba44e7e21cf62426620f4fca0b60f60/includes/class-wc-payments-account.php#L204

We can get rid of `?.` and go with just:
```js
	const getErrorMessagesFromRequirements = ( requirements ) => [
		...new Set( requirements.errors.map( ( error ) => error.reason ) ),
	];
```

Reply:
Thanks @ismaeldcom, that's a great tip! I updated in the latest commits, but I kept the `?` operators because I found if the account is not fetched properly (e.g. we can't fetch it from the server) then the page will break if we don't check for the parameters. (normally in this case we'd see the `Error determining account status` fallback).